website/docs/src/luigi-config/extended/resultRender.js

```diff
@@ -0,0 +1,177 @@
+    let output = template.render(data);
+    let searchItem = this.htmlToElement(output);
+    span.appendChild(searchItem);
+  }
+  shortText(text){
+    let short = text;
+    if (short.length > 80){
+      let lastPost = short.substring(80).indexOf(" ") + 80;
+      short = short.substring(0, lastPost)+".."
+    }
+    return this.highlightKeyword(short);
+  }
+  dataTemplate(result){
+    return {
+      label: this.shortText(result.label),
+      description: this.shortText(result.description),
+      title1: this.shortText(result.title1),
+      title2: this.shortText(result.title2),
+      title3: this.shortText(result.title3),
+    };
+  }
+  renderResult(template, data){
+    return template.render(data);
+  }
+  highlightKeyword(text){
+    this.perm(this.query).forEach(subQ => {
+      let replace = '<span class="algolia-docsearch-suggestion--highlight">'+ subQ+ '</span>';
+      text = text.replaceAll(subQ, replace);
+    })
+    return text;
+  }
+  attachItemEvents(searchItem, result){
+    searchItem.addEventListener('click', e => {
+      e.preventDefault();
+      Luigi.navigation().withParams(result.pathObject.params).navigate(result.pathObject.link);
+      Luigi.globalSearch().closeSearchResult();
+      Luigi.globalSearch().clearSearchField();
+    });
+    searchItem.addEventListener('mouseover', e => {
+      e.preventDefault();
+      searchItem.set
+      searchItem.classList.add("ds-cursor");
+
+    });
+    searchItem.addEventListener('mouseout', e => {
+      e.preventDefault();
+      searchItem.classList.remove("ds-cursor");
+    });
+  }
+  buildContainer(){
+    let span = this.createAttribute('span', {
+      "class": "algolia-autocomplete algolia-autocomplete-right",
+      "style": "position: relative; display: inline-block; direction: ltr;"});
+    let span_ = this.createAttribute('span', {
+      "class": "ds-dropdown-menu ds-with-1",
+      "role": "listbox",
+      "id" : "algolia-autocomplete-listbox-0",
+      "style": "position: absolute; top: 100%; z-index: 100; left: 0px; right: auto; display: block;"
+    });
+    let div_1 = this.createAttribute('div', {"class":"ds-dataset-1"});
+    let span_11 = this.createAttribute('span', {"class": "ds-suggestions","style": "display: block;"});
+    span.appendChild(span_);
+    span_.appendChild(div_1);
+    div_1.appendChild(span_11);
+    return {
+      container: span,
+      resultSpan : span_11
+    };
+  }
+  createAttribute(type, attributes){
+    let elem =  document.createElement(type);
+    if (!attributes){
+      return elem;
+    }
+    Object.keys(attributes).forEach(key => elem.setAttribute(key, attributes[key]))
+    return elem;
+  }
+  htmlToElement(html) {
+    return new DOMParser().parseFromString(html, "text/html").body.firstChild;
+  }
+  perm(str){
+    let results = [];
+    let arr = str.split("");
+    let len = Math.pow(arr.length, 2);
+
+    for( let i = 0; i < len; i++ ){
+      for( let k= 0, j = i; k < arr.length; k++, j >>=1){
```

Comment:
Did you mean `j >=1`? or you want to do right shift assignment

Reply:
It is fine to use >>=, otherwise the perm function will not work properly....
If you search lui... i need to highlight <b>Lui</b>gi....